### ISSUE TITLE
Publish theme to NPM

### ISSUE THREAD

**Issue by mchelen** [carlvlewis/cvStrap-jsonresume#3]
If you publish this theme to NPM with the name `jsonresume-theme-cvstrap` then anyone can use it without manually downloading it. If you want any help publishing an NPM package, feel free to let me know.

**Comment by carlvlewis** [comment]
First time publishing an NPM package, and while I'm sure I could hack it out, would love some brief guidance as I'm kind of slammed with other projects at the moment!

**Comment by carlvlewis** [comment]
Hey @mchelen, and I'm guessing I should change the name and extension of index.html to resume.template? Right?

**Comment by mchelen** [comment]
@carlvlewis Ah yep, that is the current filename used by `index.js` and would be consistent with other themes.

**Comment by carlvlewis** [comment]
So I successfully published the theme to NPM and it's appearing at http://themes.jsonresume.org/cvstrap, but it looks like i'm going to have to merge my CSS files into a single 'style.css' file in the root, which won't be impossible, but isn't ideal since I'd created separate print and screen stylesheets. Oh well... more CSS ...

**Comment by carlvlewis** [comment]
Also, @mchelen, mind checking the error returned (http://themes.jsonresume.org/cvstrap) to make sure that's the only thing I'm missing?

**Comment by mchelen** [comment]
@carlvlewis You can load multiple CSS files in `index.js` and pass them into the template. Look at how `{{{css}}}` variable is used in https://github.com/jsonresume/jsonresume-theme-boilerplate/blob/master/resume.template because you could easily have `{{{css2}}}` `{{{css3}}}` or `{{{blah}}}` as long as those variables are defined in `index.js`.

I think something is wrong with  `resume.template` because I can see things like your name & twitter profile which shouldn't be hardcoded in the template.

**Comment by carlvlewis** [comment]
yeah those need to be set as variables from JSON... i just had those in HTML then changed filename to resume.template.

**Comment by carlvlewis** [comment]
and yeah i know i can have multiple stylesheets just going to be a couple of days until i get time to do that :)

**Comment by mchelen** [comment]
Ok sure, I can do the edits to index.js if that would help.

**Comment by carlvlewis** [comment]
Hey. yeah, that'd be awesome @mchelen! I'm taking out the manually inserted fields from resume.template now and also putting the CSS in a single file. You rock!

**Comment by mchelen** [comment]
If you are combining all the CSS into one file, then the current `index.js` can be used, just put the final `style.css` in the root directory. If you prefer to have the CSS in separate files though, let me know and I'll modify `index.js`.

Also for assets like `bootstrap.js` I'd recommend using CDN versions because it simplifies the repo and makes it easier to update versions.

```
<!-- Latest compiled and minified CSS -->
<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.2/css/bootstrap.min.css">

<!-- Latest compiled and minified JavaScript -->
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.2/js/bootstrap.min.js"></script>
```

**Comment by carlvlewis** [comment]
@mchelen: Thanks for the thoroughness here! I've been busy with "real" work, so just getting to this. Yes, certainly, it makes more sense to use minified versions of the CSS and JS. Changing that now.

As for the index.js changes.which would be easier – that or compiling the CSS into a single style.css? (i know putting different stylesheets into a single file can get tedious with media@ and screen@ viewports and such, and, ideally, for clarity's sake and for users to be able to make changes quickly both to the print and screen stylesheets, I think it'd be optimal to keep them separated). Or would it be best to follow the official schema with a single style.css file in root that has the different stylesheets defined?

Let me know, and thanks again!

**Comment by carlvlewis** [comment]
Wait, couldn't the index.js changes just be as simple as calling up the location of each of the CSS files in order of priority? 

So, instead of ...
`var css = fs.readFileSync(__dirname + "/style.css", "utf-8");`
it would be...
`var css = fs.readFileSync(__dirname + "css/shared.css","css/screen.css", "css/print.css" "utf-8");`
... or something in that manner? Obviously the bootstrap.min.css too.

**Comment by carlvlewis** [comment]
Hey @mchelen, I misread your earlier post! I'd rather perform the fixes to index.js and keep the separate stylesheets. Have made fixes to resume.template.

I'll give it a go modifying `index.js` unless you happen to be free and could help out.

**Comment by mchelen** [comment]
Yup, the way described in https://github.com/carlvlewis/cvStrap/issues/3#issuecomment-71354156 would definitely work, you just need a `fs.readFileSync()` for each file:

```
var css = fs.readFileSync(__dirname + "css/shared.css", "utf-8")
+ fs.readFileSync(__dirname + "css/screen.css", "utf-8")
+ fs.readFileSync(__dirname + "css/print.css", "utf-8");
```

(added line breaks for clarity)

For the bootstrap & other standardized files, you can use https links in the `resume.template` and don't have to bother dealing with them in `index.js`.

**Comment by carlvlewis** [comment]
Awesome, just made the fixes to `resume.template` and all's good. Now just need to add those CSS calls to `index.js` and republish and that should do the trick. Dinner right now, though! Thanks, @mchelen.

**Comment by carlvlewis** [comment]
Hey @mchelen, so I added in the calls to the stylesheets to `index.js` last week -- as you can see in the file itself -- then republished to repository via NPM, but there still appears to be some CSS issues happening if you test the theme in the front-end editor. I definitely need to take out the bootstrap modal popup from the registry copy, and also re-order the elements in standard schema, as I've done in an updated version I'm calling resumeStrap. 

But I wanted to get this right first. Should I try just putting the CSS in a single file in root of project directory?
